server/src/main/java/org/elasticsearch/monitor/metrics/IndicesMetrics.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.monitor.metrics;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.lucene.store.AlreadyClosedException;
+import org.elasticsearch.cluster.routing.ShardRouting;
+import org.elasticsearch.common.component.AbstractLifecycleComponent;
+import org.elasticsearch.common.util.SingleObjectCache;
+import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.index.IndexMode;
+import org.elasticsearch.index.IndexService;
+import org.elasticsearch.index.shard.IllegalIndexShardStateException;
+import org.elasticsearch.index.shard.IndexShard;
+import org.elasticsearch.indices.IndicesService;
+import org.elasticsearch.telemetry.metric.LongWithAttributes;
+import org.elasticsearch.telemetry.metric.MeterRegistry;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.EnumMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * {@link IndicesMetrics} monitors index statistics on an Elasticsearch node and exposes them as metrics
+ * through the provided {@link MeterRegistry}. It tracks the current total number of indices, document count, and
+ * store size (in bytes) for each index mode.
+ */
+public class IndicesMetrics extends AbstractLifecycleComponent {
+    private final Logger logger = LogManager.getLogger(IndicesMetrics.class);
+    private final MeterRegistry registry;
+    private final List<AutoCloseable> metrics = new ArrayList<>();
+    private final IndicesStatsCache stateCache;
+
+    public IndicesMetrics(MeterRegistry meterRegistry, IndicesService indicesService, TimeValue metricsInterval) {
+        this.registry = meterRegistry;
+        var cacheExpiry = new TimeValue(metricsInterval.getMillis() / 2);
```

Comment:
Why `metricsInterval.getMillis() / 2`? Is the intent that each time metric interval has fresh metrics?

Reply:
I've added a comment in https://github.com/elastic/elasticsearch/pull/113737/commits/9523b6e8ee5b13dfe4e894e999d3a919e2a52969.